src/cli/subcommands/car_cmd.rs

```diff
@@ -0,0 +1,77 @@
+// Copyright 2019-2023 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use std::path::PathBuf;
+
+use clap::Subcommand;
+use fvm_ipld_car::Block;
+use fvm_ipld_car::CarHeader;
+use fvm_ipld_car::CarReader;
+use tokio_util::compat::TokioAsyncReadCompatExt;
+
+use crate::ipld::CidHashSet;
+
+#[derive(Debug, Subcommand)]
+pub enum CarCommands {
+    Concat {
+        first: PathBuf,
+        second: PathBuf,
+        /// The output `.car` file path
+        #[arg(short, long)]
+        output: PathBuf,
+    },
+}
+
+impl CarCommands {
+    pub async fn run(self) -> anyhow::Result<()> {
+        match self {
+            Self::Concat {
+                first,
+                second,
+                output,
+            } => {
+                let reader_a = CarReader::new(tokio::fs::File::open(first).await?.compat()).await?;
+                let reader_b =
+                    CarReader::new(tokio::fs::File::open(second).await?.compat()).await?;
```

Comment:
`CarReader` isn't buffered. Adding a `BufReader` should give you much better performance.

Reply:
Fixed